### DISCUSSION TITLE
Negating a child in a query?

### DISCUSSION THREAD

**Discussion by bennypowers** [nvim-treesitter/nvim-treesitter#2582] [Q&A] [answered]
Please see this stack overflow: https://stackoverflow.com/questions/71270346/negating-a-child-node-in-a-tree-sitter-query

Given a tree for the following HTML:
```html
<script>console.log('should parse as js')</script>
<script async defer>console.log('works')</script>
```
How can I query for the first element, i.e. a script_element which has no attributes at all

This query fails with `invalid field`

```lisp
(script_element
   (start_tag !attribute))
```

Related: #2577

**Comment by stsewd** [comment, marked as answer]
A query like this should work

```
(script_element (start_tag (tag_name) .)) @foo
```


The negation operator works only on fields, here `attribute` is a node.
My query tries to match a script element where the name is the final children of the `start_tag` node, e.g, it has not attributes.

**Reply by bennypowers** [reply]
Thanks!

**Comment by dchinmay2** [comment]
is there a way to assert that the node i'm capturing doesn't have children?
